src/main/java/org/amahi/anywhere/service/AudioService.java

```diff
@@ -144,15 +153,18 @@ private void setUpBus() {
     }
 
     private void setUpAudioPlayer() {
-        audioPlayer = new MediaPlayer();
-
-        audioPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
-        audioPlayer.setWakeMode(this, PowerManager.PARTIAL_WAKE_LOCK);
-        audioPlayer.setVolume(1.0f, 1.0f);
+        audioPlayer =
+            ExoPlayerFactory.newSimpleInstance(
+                getApplicationContext(), new DefaultTrackSelector(), new DefaultLoadControl());
+        audioPlayer.addListener(this);
+        audioPlayer.setPlayWhenReady(true);
+        audioPlayer.setVolume(1.0f);
+    }
 
-        audioPlayer.setOnPreparedListener(this);
-        audioPlayer.setOnCompletionListener(this);
-        audioPlayer.setOnErrorListener(this);
+    private MediaSource buildMediaSource(Uri uri) {
+        return new ExtractorMediaSource.Factory(
+            new DefaultHttpDataSourceFactory(getString(R.string.application_name)))
```

Comment:
Instead of using `application_name` as user-agent, you can use `Identifier.getUserAgent(context)`.